app/src/main/java/com/kickstarter/viewmodels/SetPasswordViewModel.kt

```diff
@@ -93,65 +99,69 @@ interface SetPasswordViewModel {
 
             setNewPassword
                 .map { it.warning() }
+                .filter { ObjectUtils.isNotNull(it) }
+                .map { requireNotNull(it) }
+                .filter { it != 0 }
```

Comment:
I had to update the function `warning()` to return a non-null value, as it was still causing the app to crash due to a the mapper function returning a null value, even with the `isNotNull` checks in the chain. I'm not exactly sure why this is the case here or what makes this different, as opposed to other places where a null check in the chain is enough to not cause a crash (the intent rx chain above is an example of this, where the getStringExtra returns a null value if there is no extra with that key attached to the intent, but it does not crash.)